src/lib.rs

```diff
@@ -402,21 +402,46 @@ impl<T> Drop for DiplomaticBag<T> {
     }
 }
 
-/// Unfortunately, we can't implement `Debug` properly for `T`s that implement
-/// `Debug` because the formatter isn't `Send`. Instead, we just be a bit
-/// enigmatic and say we have something but refuse to say what it is.
-impl<T> fmt::Debug for DiplomaticBag<T> {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "DiplomaticBag<{}> {{ .. }}", std::any::type_name::<T>())
+/// This `Debug` impl currently won't forward any formatting specifiers except
+/// for the `alternate` specifier (`#`).
+impl<T: fmt::Debug> fmt::Debug for DiplomaticBag<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        struct AsIs(String);
+
+        impl fmt::Debug for AsIs {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                f.write_str(&self.0)
+            }
+        }
+
+        let alt = f.alternate();
+
+        // Basically, run Debug on the worker thread, then send the resulting String back
+        let res = self.as_ref()
+            .map(move |_, this| {
+                if alt {
+                    format!("{:#?}", this)
+                } else {
+                    format!("{:?}", this)
+                }
+            })
+            .into_inner();
+
+        let res = AsIs(res); // So that we don't format it with quotes, etc.
+
+        f.debug_struct("DiplomaticBag")
+            .field("inner", &res)
+            .finish()
     }
 }
 
 // We can, however, implement a bunch of other useful standard traits, as below.
-// Annoyingly, `Display`, `serde::Serialize`, and `serde::Deserialize` all
-// suffer from the same problem as `Debug`. `From`, and `TryFrom` fail due
-// to the orphan rules. `AsRef` and `AsMut` can be implemented but conflict with
-// the existing `as_ref` and `as_mut` methods, and I'm not sure the confusion is
-// worth it.
+// Annoyingly, `serde::Serialize`, and `serde::Deserialize` both suffer from a worse
+// problem than `Debug` does. They have over 30 methods that would each need to
```

Comment:
I took another punt at implementing `Serialize` I agree that it's possible but a nightmare. Can you please keep the mention of `Display` here? `Display` is worse because we must preserve all the formatting specifiers.

Reply:
@Skepfyr 
Did you want to have an implementation of Display like Debug and just mention that only the `#` formatting specifier works?

Reply:
No. The Debug implementation can be a best effort thing as it should only be used for debugging. I'd want a Display implementation to be perfect as it's intended for non-technical users.

Reply:
That should be fixed now in the notes